src/main/java/seedu/address/logic/parser/AddStudCommandParser.java

```diff
@@ -35,23 +35,37 @@ public class AddStudCommandParser implements Parser<AddStudCommand> {
     public AddStudCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
                 ArgumentTokenizer.tokenize(args, PREFIX_STUDENT_NAME, PREFIX_ID, PREFIX_CLASS,
-                        PREFIX_PARENT_NAME, PREFIX_PHONE, PREFIX_ADDRESS, PREFIX_TAG);
+                        PREFIX_PARENT_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_TAG);
 
-        if (!arePrefixesPresent(argMultimap, PREFIX_STUDENT_NAME, PREFIX_ID, PREFIX_CLASS, PREFIX_PARENT_NAME,
-                PREFIX_PHONE, PREFIX_ADDRESS)
+        if (!arePrefixesPresent(argMultimap, PREFIX_STUDENT_NAME, PREFIX_ID, PREFIX_CLASS)
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddStudCommand.MESSAGE_USAGE));
         }
 
         Name studentName = ParserUtil.parseName(argMultimap.getValue(PREFIX_STUDENT_NAME).get());
         Id id = ParserUtil.parseId(argMultimap.getValue(PREFIX_ID).get());
         Class className = ParserUtil.parseClass(argMultimap.getValue(PREFIX_CLASS).get());
-        Name parentName = ParserUtil.parseName(argMultimap.getValue(PREFIX_PARENT_NAME).get());
-        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-        Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
+        Phone phone;
+        Name parentName;
+        Email email;
+        if (argMultimap.getValue(PREFIX_PHONE).isEmpty()) {
+            phone = new Phone();
+        } else {
+            phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).orElse(""));
+        }
+        if (argMultimap.getValue(PREFIX_PARENT_NAME).isEmpty()) {
+            parentName = new Name();
+        } else {
+            parentName = ParserUtil.parseName(argMultimap.getValue(PREFIX_PARENT_NAME).orElse(""));
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isEmpty()) {
+            email = new Email();
+        } else {
+            email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).orElse(""));
+        }
```

Comment:
Might be redundant to do an `isEmpty()` check and then use `orElse()` for the Optional value. Probably can handle the empty input in the `ParserUtil` class.

Reply:
Have removed the `orElse` to simply a `get`.
The empty input is handled here to distinguish between when a user does not enter a flag, where we are skipping the check and storing an empty string (calling constructor without check),
and when a user enters a blank value with the flag, where `argMultimap.getValue` will return an empty string (this will be subjected to the validity check in the parserUtil class).

Reply:
👍🏻 yup was thinking if there was a better way but I think this is good. Thanks for clarifying!